conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/UndertowRuntime.java

```diff
@@ -41,4 +43,8 @@
 
     /** Provides the {@link ExceptionHandler} used to map {@link Throwable failures} to Conjure responses. */
     ExceptionHandler exceptionHandler();
+
+    /** Provides the {@link Contexts} which safely expose request details to resources. */
+    @Beta
+    Contexts contexts();
```

Comment:
Why this layer of indirection? Why can't you return RequestContext directly, but only create it on first call, to maintain laziness?

Reply:
Ok, I understand: genned code impls HttpHandler. UndertowRuntime is a per-server factory for stuff that the codegen code needs.